Accept contributions! 💝

TedTed opened the pull request:
Add more comments in the Makefile, and make CONTRIBUTING.md friendlier and into a minimal guide to local development.

Maegereg commented on CONTRIBUTING.md:
```diff
@@ -1,21 +1,54 @@
-To work locally, install Poetry, and then install our dev dependencies by running `poetry install` from the root of this repository.
+### Installation
+
+We recommend using a Python virtual environment to do local development, for example using [pyenv](https://github.com/pyenv/pyenv): it is useful to make sure you're working with a supported Python version, and minimize interactions with your system Python environment.
```
```suggestion
We recommend using the lowest supported python version for local development. Tools like [pyenv](https://github.com/pyenv/pyenv) may be useful for managing your python version.
```

pyenv doesn't actually manage virtual environments, despite the name. It just managed installed python versions. Poetry handles creating the virtual environments.

At some point (soon?) we probably want to switch these instructions to recommend `uv` instead (cc @tmager ), but I don't see a super easy way to substitute `uv` for `pyenv`.

Maegereg commented on CONTRIBUTING.md:
```diff
@@ -1,21 +1,54 @@
+
+Note that some operating systems, including macOS, include versions of `make` that are too old to run this project's [Makefile](./Makefile) correctly. macOS users should [install a newer version of make using Homebrew](https://formulae.brew.sh/formula/make#default).
+
+Behind the scenes, these commands use the Poetry environment, and rely on [nox](https://nox.thea.codes/en/stable/index.html) for test automation. For more information about how it works, see our [nox-utils](https://github.com/opendp/tumult-tools/tree/main/nox-utils) repository.
```
```suggestion
Behind the scenes, these commands use the Poetry environment, and rely on [nox](https://nox.thea.codes/en/stable/index.html) for test automation. You can get a bit more fine-grained control and access additional tools by running nox commands directly (see [this tutorial](https://nox.thea.codes/en/stable/tutorial.html)). For a list of available nox sessions, you can run `poetry run nox --list`.
```

Maegereg reviewed (commented):
I also left a comment on the associated issue.

TedTed commented:
I simplified the Makefile as suggested on the issue, and added more instructions in CONTRIBUTING.md to give people a basic idea of how to use nox. I moved together both kinds of doctests (docstrings and RST files) under `make test` so that `make docs` is still reasonably fast.

One thing that I'd like to add later: instructions on how to run a small subset of tests. Currently, on my machine, `make test` needs more than 30 minutes to run…

TedTed commented on CONTRIBUTING.md:
```diff
@@ -1,21 +1,54 @@
+We recommend using the lowest supported Python version for local development. Tools like [pyenv](https://github.com/pyenv/pyenv) may be useful for managing your Python version.
 
-See the [installation instructions](https://docs.tmlt.dev/analytics/latest/installation.html#installation-instructions) for more information about prerequisites.
+We use [Poetry](https://python-poetry.org/) for dependency management during development. To set up your environment, start by installing the prerequisites listed in the [installation instructions](https://docs.tmlt.dev/platform/latest/installation.html#prerequisites), then install Poetry using `pip install poetry`, and finally install our dev dependencies by running `poetry install` from the roo ...
```
Done!
